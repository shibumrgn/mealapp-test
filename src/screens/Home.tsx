import { View, Text, ScrollView } from 'react-native'
import React from 'react'
import { Brand } from '../constants/Brand';
import CustomInput from '../components/CustomInput';
import TrendingNow from '../components/TrendingNow';
import PopularCategory from '../components/PopularCategory';

const Home = () => {
  return (
    <ScrollView style={{
        flex: 1,
        paddingBottom: 100,
        marginBottom: 60
    }}>
        <View
            style={{
                padding: 20
            }}
        >
            <Text
                style={{
                    ...Brand.header,
                    marginBottom: 20
                }}
            >
                    Good afternoon,{"\n"}Guest!
            </Text>
            <CustomInput
                iconName='search-outline'
                placeholder='Search Recipes'
            />
        </View>
        <TrendingNow/>
        <PopularCategory/>
    </ScrollView>
  )
}

export default Home;