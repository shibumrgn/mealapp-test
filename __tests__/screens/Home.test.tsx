import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../../src/screens/Home';
jest.useFakeTimers();
jest.mock('axios');

it('renders correctly', () => {
    const render = renderer.create(<Home />)?.toJSON();
    expect(render).toMatchSnapshot();
});
