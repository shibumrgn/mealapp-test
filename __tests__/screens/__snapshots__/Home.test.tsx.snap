// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "flex": 1,
      "marginBottom": 60,
      "paddingBottom": 100,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "padding": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 30,
            "fontWeight": "bold",
            "marginBottom": 20,
          }
        }
      >
        Good afternoon,
        

        Guest!
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "lightgrey",
            "borderRadius": 10,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "marginBottom": 10,
            "paddingHorizontal": 10,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          ï”‰
        </Text>
        <TextInput
          placeholder="Search Recipes"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "marginBottom": 30,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 30,
            "fontWeight": "bold",
            "marginBottom": 20,
            "marginLeft": 20,
          }
        }
      >
        Trending Now ðŸ”¥
      </Text>
      <RCTScrollView
        horizontal={true}
        showsHorizontalScrollIndicator={false}
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "paddingLeft": 20,
            "width": "100%",
          }
        }
      >
        <View />
      </RCTScrollView>
    </View>
    <View
      style={
        Object {
          "marginBottom": 30,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 30,
            "fontWeight": "bold",
            "marginBottom": 20,
            "marginLeft": 20,
          }
        }
      >
        Popular Category
      </Text>
      <RCTScrollView
        horizontal={true}
        showsHorizontalScrollIndicator={false}
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "paddingLeft": 20,
            "width": "100%",
          }
        }
      >
        <View />
      </RCTScrollView>
    </View>
    <View
      style={
        Object {
          "marginBottom": 30,
        }
      }
    >
      <RCTScrollView
        horizontal={true}
        showsHorizontalScrollIndicator={false}
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "paddingLeft": 20,
            "width": "100%",
          }
        }
      >
        <View />
      </RCTScrollView>
    </View>
  </View>
</RCTScrollView>
`;
